docs/specs/editorconfig-in-the-compiler.md

```diff
@@ -0,0 +1,111 @@
+All-capital words are interpreted per [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
```

Comment:
:arrow_right: As a quick follow-up, based on further conversation with Jared I believe his intent here was referring to a potential ability to enable or disable some diagnostic (toggle the severity between None/Hidden/Info for "off", or Warning/Error for "on"). Due avoid even the possibility of compatibility problems, it was decided that this functionality would not be added at this time. This could be seen as limiting for some use cases, but offered substantial flexibility to improve the overall user experience in several others.

* What we "gave up":
  * Ability to change severity of warnings produced by the C# compiler (the ones that start with **CS** and exist even without analyzers)
* What we "gained":
  * Ability to enable .editorconfig lookup by default, so analyzer authors targeting supported versions can use it for analyzer-specific configuration
  * Consistent behavior out-of-the-box for in-IDE experience (editing, analyzers, etc.) and build - for users who have analyzers installed as NuGet packages they will use .editorconfig in the same manner without requiring users to manually configure the project to behave consistently